crates/swc_css_parser/src/parser/at_rule.rs

```diff
@@ -1324,27 +1367,22 @@ where
     fn parse(&mut self) -> PResult<PageRule> {
         let start = self.input.cur_span()?.lo;
 
-        let prelude = {
-            let mut items = vec![];
-            loop {
-                self.input.skip_ws()?;
-
-                if is!(self, "{") {
-                    break;
-                }
+        let prelude = if !is!(self, "{") {
+            Some(self.parse()?)
+        } else {
+            None
+        };
 
-                items.push(self.parse()?);
+        expect!(self, "{");
 
-                self.input.skip_ws()?;
-                if !eat!(self, ",") {
-                    break;
-                }
-            }
-            items
-        };
+        self.ctx.in_page_at_rule = true;
 
         let block = self.parse()?;
 
+        self.ctx.in_page_at_rule = false;
```

Comment:
I think we can use `.with_ctx` instead?
`with_ctx` returns a RAII guard which restores the original context on drop.

Reply:
```rust
let ctx = Ctx {
    in_page_at_rule: true,
    ..self.ctx
};
let block = self.with_ctx(ctx).parse()?;
```

will do the trick.